src/Tuple.cpp

```diff
@@ -31,42 +59,52 @@ TetMesh::Tuple TetMesh::Tuple::init_from_tet(const TetMesh& m, int tid){
 
 bool TetMesh::Tuple::is_valid(const TetMesh& m) const
 {
-    if (m.m_vertex_connectivity[m_vid].m_is_removed ||
-        m.m_tet_connectivity[m_tid].m_is_removed)
+    if (m.m_vertex_connectivity[m_vid].m_is_removed || m.m_tet_connectivity[m_tid].m_is_removed)
         return false;
     return true;
 }
 
 void TetMesh::Tuple::update_version_number(const TetMesh& m)
 {
-    assert(m_timestamp >= m.m_tet_connectivity[m_tid].timestamp);
+    assert(m_timestamp < m.m_tet_connectivity[m_tid].timestamp);
     m_timestamp = m.m_tet_connectivity[m_tid].timestamp;
 }
 
-int TetMesh::Tuple::get_version_number() { return m_timestamp; }
+int TetMesh::Tuple::get_version_number()
+{
+    return m_timestamp;
+}
 
 bool TetMesh::Tuple::is_version_number_valid(const TetMesh& m) const
 {
     if (m_timestamp != m.m_tet_connectivity[m_tid].timestamp) return false;
     return true;
 }
 
-void TetMesh::Tuple::print_info() { logger().trace("tuple: {} {} {} {}", m_vid, m_eid, m_fid, m_tid); }
+void TetMesh::Tuple::print_info()
+{
+    logger().trace("tuple: {} {} {} {}", m_vid, m_eid, m_fid, m_tid);
+}
 
-size_t TetMesh::Tuple::vid() const { return m_vid; } // update eid and fid
+size_t TetMesh::Tuple::vid() const
+{
+    return m_vid;
+} // update eid and fid
 
 size_t TetMesh::Tuple::eid(const TetMesh& m) const
 {
     int v1_id = m.m_tet_connectivity[m_tid][m_local_edges[m_eid][0]];
     int v2_id = m.m_tet_connectivity[m_tid][m_local_edges[m_eid][1]];
-    auto n12_t_ids = set_intersection(m.m_vertex_connectivity[v1_id].m_conn_tets,
-                                      m.m_vertex_connectivity[v2_id].m_conn_tets);
+    auto n12_t_ids = set_intersection(
+        m.m_vertex_connectivity[v1_id].m_conn_tets,
+        m.m_vertex_connectivity[v2_id].m_conn_tets);
+    assert(!n12_t_ids.empty());
+
     int tid = *std::min_element(n12_t_ids.begin(), n12_t_ids.end());
-    for(int j=0;j<6;j++){
+    for (int j = 0; j < 6; j++) {
         int tmp_v1_id = m.m_tet_connectivity[tid][m_local_edges[j][0]];
         int tmp_v2_id = m.m_tet_connectivity[tid][m_local_edges[j][1]];
-        if(tmp_v1_id == v1_id && tmp_v2_id == v2_id
-            ||tmp_v1_id == v2_id && tmp_v2_id == v1_id)
+        if (tmp_v1_id == v1_id && tmp_v2_id == v2_id || tmp_v1_id == v2_id && tmp_v2_id == v1_id)
```

Comment:
I would recommend add parenthesize for readability.
```suggestion
        if ((tmp_v1_id == v1_id && tmp_v2_id == v2_id) || (tmp_v1_id == v2_id && tmp_v2_id == v1_id))
```